grype/pkg/package.go

```diff
@@ -56,11 +57,24 @@ func New(p pkg.Package) Package {
 	}
 }
 
-func FromCatalog(catalog *pkg.Catalog) []Package {
+func FromCatalog(catalog *pkg.Catalog, config ProviderConfig) []Package {
 	result := make([]Package, 0, catalog.PackageCount())
+	missingCPEs := false
 	for _, p := range catalog.Sorted() {
+		if len(p.CPEs) == 0 {
+			// For SPDX (or any format, really) we may have no CPEs
+			if config.GenerateMissingCPEs {
+				p.CPEs = cpe.Generate(p)
+			} else {
+				log.Debugf("No CPEs for package: %s", p)
```

Comment:
nit: lowercase for consistency
```suggestion
				log.Debugf("no CPEs for package: %s", p)
```